libvirt/resource_libvirt_pool.go

```diff
@@ -99,17 +109,56 @@ func resourceLibvirtPoolCreate(d *schema.ResourceData, meta interface{}) error {
 	log.Printf("[DEBUG] Pool with name '%s' does not exist yet", poolName)
 
 	poolPath := d.Get("path").(string)
-	if poolPath == "" {
-		return fmt.Errorf("\"path\" attribute is requires for storage pools of type \"dir\"")
-	}
 
-	poolDef := libvirtxml.StoragePool{
-		Type: "dir",
-		Name: poolName,
-		Target: &libvirtxml.StoragePoolTarget{
-			Path: poolPath,
-		},
+	var poolDef *libvirtxml.StoragePool
+	// In some cases we don't need to build the pool
+	needToBuild := true
```

Comment:
Is this really needed with the `VIR_STORAGE_POOL_BUILD_NO_OVERWRITE` flag? eg. [see here](https://github.com/libvirt/libvirt/blob/master/src/storage/storage_backend_logical.c).